source/common/protobuf/create_reflectable_message.cc

```diff
@@ -0,0 +1,426 @@
+#include "envoy/service/discovery/v3/discovery_descriptor.pb.h"
+#include "envoy/service/endpoint/v3/eds_descriptor.pb.h"
+#include "envoy/service/endpoint/v3/leds_descriptor.pb.h"
+#include "envoy/service/extension/v3/config_discovery_descriptor.pb.h"
+#include "envoy/service/health/v3/hds_descriptor.pb.h"
+#include "envoy/service/listener/v3/lds_descriptor.pb.h"
+#include "envoy/service/load_stats/v3/lrs_descriptor.pb.h"
+#include "envoy/service/metrics/v3/metrics_service_descriptor.pb.h"
+#include "envoy/service/ratelimit/v3/rls_descriptor.pb.h"
+#include "envoy/service/route/v3/rds_descriptor.pb.h"
+#include "envoy/service/route/v3/srds_descriptor.pb.h"
+#include "envoy/service/runtime/v3/rtds_descriptor.pb.h"
+#include "envoy/service/secret/v3/sds_descriptor.pb.h"
+#include "envoy/type/matcher/metadata_descriptor.pb.h"
+#include "envoy/type/matcher/node_descriptor.pb.h"
+#include "envoy/type/matcher/number_descriptor.pb.h"
+#include "envoy/type/matcher/path_descriptor.pb.h"
+#include "envoy/type/matcher/regex_descriptor.pb.h"
+#include "envoy/type/matcher/string_descriptor.pb.h"
+#include "envoy/type/matcher/struct_descriptor.pb.h"
+#include "envoy/type/matcher/value_descriptor.pb.h"
+#include "envoy/type/matcher/v3/filter_state_descriptor.pb.h"
+#include "envoy/type/matcher/v3/http_inputs_descriptor.pb.h"
+#include "envoy/type/matcher/v3/metadata_descriptor.pb.h"
+#include "envoy/type/matcher/v3/node_descriptor.pb.h"
+#include "envoy/type/matcher/v3/number_descriptor.pb.h"
+#include "envoy/type/matcher/v3/path_descriptor.pb.h"
+#include "envoy/type/matcher/v3/regex_descriptor.pb.h"
+#include "envoy/type/matcher/v3/status_code_input_descriptor.pb.h"
+#include "envoy/type/matcher/v3/string_descriptor.pb.h"
+#include "envoy/type/matcher/v3/struct_descriptor.pb.h"
+#include "envoy/type/matcher/v3/value_descriptor.pb.h"
+#include "envoy/type/metadata/v3/metadata_descriptor.pb.h"
+#include "envoy/type/tracing/v3/custom_tag_descriptor.pb.h"
+#include "envoy/type/v3/hash_policy_descriptor.pb.h"
+#include "envoy/type/v3/http_descriptor.pb.h"
+#include "envoy/type/v3/http_status_descriptor.pb.h"
+#include "envoy/type/v3/percent_descriptor.pb.h"
+#include "envoy/type/v3/range_descriptor.pb.h"
+#include "envoy/type/v3/ratelimit_strategy_descriptor.pb.h"
+#include "envoy/type/v3/ratelimit_unit_descriptor.pb.h"
+#include "envoy/type/v3/semantic_version_descriptor.pb.h"
+#include "envoy/type/v3/token_bucket_descriptor.pb.h"
+#include "envoy/watchdog/v3/abort_action_descriptor.pb.h"
+
+using ::cc_proto_descriptor_library::internal::FileDescriptorInfo;
+
+namespace {
+
+std::unique_ptr<cc_proto_descriptor_library::TextFormatTranscoder> createTranscoder() {
+  std::unique_ptr<cc_proto_descriptor_library::TextFormatTranscoder> transcoder =
+      std::make_unique<cc_proto_descriptor_library::TextFormatTranscoder>();
+  std::vector<FileDescriptorInfo> file_descriptors = {
+      protobuf::reflection::envoy_config_core_v3_base::kFileDescriptorInfo,
```

Comment:
What would happen if a new proto file was added but this list was left unmodified?

Reply:
Discussed offline. The client integration test in the mobile compile time options build would fail.